front/tests/utils/api-utils.ts

```diff
@@ -59,6 +60,35 @@ export const getApiRequest = async <T>(
   return response.json() as T;
 };
 
+/**
+ * Send GET requests to the specified paginated listing API endpoint with optional query parameters.
+ * This function will send as many requests as needed to get all pages, and return the aggregated full list of items.
+ * To get a specific page, use getApiRequest with appropriate pagination parameters.
+ *
+ * @template T - Type of the listed item.
+ * @param url - The API endpoint URL.
+ * @param params - Optional query parameters to include in the request.
+ */
+export const listApiRequest = async <T>(
+  url: string,
+  params?: { [key: string]: string | number | boolean }
```

Comment:
Probably `Record<string, unknown>` is enough here? In other words, we don't care about the type of the values.

Reply:
I don't mind either way, I didn't realize this was carried over from elsewhere.